misc-tools/import-contest.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env python3
+
+#
+# import-contest.py -- Convenience script to import a contest (including metadata, teams and
+# problems) via the command line. Supports reading credentials from ~/.netrc. See also
+# https://www.domjudge.org/docs/manual/main/import.html
+#
+# Part of the DOMjudge Programming Contest Jury System and licensed
+# under the GNU GPL. See README and COPYING for details.
+#
+
+import fileinput
+import json
+import os.path
+import requests
+import requests.utils
+import sys
+from typing import List
+import yaml
+
+headers = {
+    "user-agent": f"domjudge-import-contest ({requests.utils.default_user_agent()})"}
+
+api_url = None
+cid = None
+
+
+def confirm(message: str, default: bool) -> bool:
+    answer = 'x'
+    while answer not in ["", "y", "n"]:
+        yn = "Y/n" if default else "y/N"
+        answer = input(f"{message} ({yn}) ").lower()
+        if answer == "":
+            answer = "y" if default else "n"
+    return answer == "y"
+
+
+def usage():
+    print(f"Usage '{sys.argv[0]} <domjudge-api-url>")
+    exit(1)
+
+
+def do_api_request(name: str):
+    '''Perform an API call to the given endpoint and return its data.
+
+    Parameters:
+        name (str): the endpoint to call
+
+    Returns:
+        The endpoint contents.
+
+    Raises:
+        RuntimeError when the response is not JSON or the HTTP status code is non 2xx.
+    '''
+
+    url = f'{api_url}/{name}'
+
+    try:
+        response = requests.get(url, headers=headers)
+    except requests.exceptions.RequestException as e:
+        raise RuntimeError(e)
+
+    if response.status_code >= 300:
+        print(response.text)
+        if response.status_code == 401:
+            raise RuntimeError(
+                'Authentication failed, please check your DOMjudge credentials in ~/.netrc.')
+        else:
+            raise RuntimeError(
+                f'API request {name} failed (code {response.status_code}).')
```

Comment:
This is duplicated

Reply:
I will put the “parsing” in a shared method